core/trie.go

```diff
@@ -144,6 +146,21 @@ func NewTrie(storage TrieStorage, height uint) *Trie {
 	}
 }
 
+// DoOnTempTrie creates an in-memory Trie of height `height` and runs `do` on that Trie
+func DoOnTempTrie(height uint, do func(*Trie) error) error {
+	db, err := db.NewInMemoryDb()
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+
+	txn := db.NewTransaction(true)
+	defer txn.Discard()
+
+	trieTxn := NewTrieTxn(txn, nil)
+	return do(NewTrie(trieTxn, height))
+}
+
```

Comment:
It should be moved to the test file, right?

Reply:
Not really, this is useful for transaction and event commitments as well. We will need temporary tries to build the commitment.